### ISSUE TITLE
Custom CSS Box for Admins

### ISSUE THREAD

**Issue by tiotrom** [friendica/friendica#9903]
Like PeerTube, it would be useful for admins to have the ability to easily inject CSS and customize a bit the instance.

**Comment by SpcCw** [comment]
Yep, it could be cool.

Basically it already can be done by making your own copy of theme scheme and editing CSS file. I have a few tweaks done this way. But it requires editing files on the server. On the other hand CSS changes are rather advanced stuff too.

**Comment by tiotrom** [comment]
Which file more exactly? I am using the Frio theme? And won't the changes be overwritten by updates?

**Comment by MrPetovan** [comment]
You're absolutely right, there's no safe way of adding custom CSS in Friendica yet. But if you're adventurous, you can pick a Frio scheme and start modifying the corresponding scheme file (in `view/theme/frio/scheme`) to your liking. It will affect all users selecting the same scheme.

**Comment by tiotrom** [comment]
Ok so there is no good practice here, for now ;). Would it be difficult to add such a feature for the admin panel? This would allow people to customize their instances a lot.

**Comment by MrPetovan** [comment]
No, not that difficult, we already have dynamic CSS sheets, they can be leverage with a custom CSS field. It is rendered slightly complicated by the existence of themes and schemes though. Should we allow separate custom CSS for each theme? Each scheme?

**Comment by tiotrom** [comment]
I think a simple CSS box like Peertube has, independent of the theme, for those who know how to deal with CSS. For me would be fantastic since I can easily inspect the page and add CSS to tweak the looks as I want. I'd absolutely love that. Just a simple injecting-CSS box is what I'm thinking about.

**Comment by MrPetovan** [comment]
What about the potential side-effect depending on the theme each user selected? Does PeerTube have a per-user theme selector?

**Comment by SpcCw** [comment]
@tiotrom server administrator can create their own "schemes" for Frio theme which are then selectable by users.

These files are located in: `/view/theme/frio/scheme/` and there are two files for each scheme: php and css. You can clone existing ones, peek inside th get some ideas. For example I inject this to make Frio look better on large screens:

```
@media only screen and (min-width: 1700px) {
    .container {
        width: 68%;
        }
}
```

Custom color schemes aren't overwritten with updates (not happened yet at least) :)

**Comment by MrPetovan** [comment]
This is definitely a possibility as well with no potential side-effects on the rest of the themes/schemes.

**Comment by tiotrom** [comment]
> What about the potential side-effect depending on the theme each user selected? Does PeerTube have a per-user theme selector?

I  checked and doesn't seem to have them per theme. But a general box. 

@SpcCw  thanks! What if you activate a CSS box per theme? Like Say the Frio theme is selected by the admin, you get a CSS bos for the Frio theme. If another theme is selected then a different CSS box will appear. Or, add 4 or so boxes for each theme, and is up to the admins to use it wisely.

I could clone and tweak but am afraid in the long run is not sustainable. A CSS injection injects CSS rather than having the entire CSS file of a theme that can change with updates. Say I clone Frio with its css file, then I customize it. The frio gets updated and I'd like to use the latest version but not I clone the new frio again and try to see what changes to do to the css? Do you know what I mean?

**Comment by MrPetovan** [comment]
@SpcCw was suggesting you to create a new empty file in the `/view/theme/frio/scheme/` folder and fill it with your modifications. Then you would be able to select it as the node scheme which would be shown by default to users who didn't change their theme/scheme.

**Comment by tiotrom** [comment]
Ah ok I get it. Beautiful. Had no clue! Thanks.
I looked through them a bit, I am very familiar with CSS, but how do I make mine as default?

**Comment by MrPetovan** [comment]
Default for the node? You have to select the scheme once you created the file in the admin settings -> General Information -> Default theme settings.

**Comment by SpcCw** [comment]
@tiotrom also you don't have to change everything, just what you want to be different from defaults. Then these custom scheme files remain. So far I haven't needed to fix any settings after updates.

Yes, in theory if Frio undergoes massive changes your custom CSS can become invalid. E.g. I have custom color for unseen notifications `li.notification-unseen` but what if the class changes name? Surely custom CSS will need updating. But I am pretty sure same would apply for any CSS customization.

**Comment by tiotrom** [comment]
Ok I managed to create my own scheme and to edit it. Very cool! But I still cannot see how I can select it for the entire instance from the admin panel...
![image](https://user-images.githubusercontent.com/34804533/107131153-5e564680-68d4-11eb-81ab-1605fe9a41af.png)

I can only select Frio but not a particular Frio scheme... What am I missing?

**Comment by MrPetovan** [comment]
"Change default theme settings" in your screenshot.

**Comment by tiotrom** [comment]
Duh....indeed. You know what confused me a lot? I clicked that and saw this:
![image](https://user-images.githubusercontent.com/34804533/107134379-d2531780-68f1-11eb-90d4-667ca39b173b.png)
And I thought for sure that the two are images - like showcasing how the theme looks like. But apparently the second is a setting....

**Comment by MrPetovan** [comment]
Ha ha ha, sorry for the confusion!

**Comment by tiotrom** [comment]
So I played with the Peertube CSS box and is really useful. How does it deal with other themes? Overrides them...and it is the admin's responsibility to make sure it is compatible with any. Also, if use selects another theme, it should ignore the CSS box I'd say. The CSS box should only work with the default theme chosen by the admin I'd say. But in any case, it is super useful.
